src/Architecture.cpp

```diff
@@ -97,7 +97,7 @@ void Architecture::loadProperties(const std::string& filename) {
 }
 
 void Architecture::loadProperties(std::istream&& is) {
-    static auto SingleQubitGates = {"id", "u1", "u2", "u3", "rz", "sx", "x"};
+    static const std::vector<std::string> SingleQubitGates = {"id", "u1", "u2", "u3", "rz", "sx", "x"};
```

Comment:
any particular reason for this change? I mean, the addition of `const` totally makes sense, but why drop the `auto`?

Reply:
The auto made problems with some types of compilers, because it was converted to a vector of char* and this in turn only looked at the first character when doing a conversion to a gate. 
For gcc its fine, but msvc complained for some versions of the compiler. So I thought it best to be more clear which type is wanted.